gensim/models/fasttext.py

```diff
@@ -41,7 +53,49 @@
     >>> fname = get_tmpfile("fasttext.model")
     >>>
     >>> model.save(fname)
-    >>> model = FastText.load(fname)  # you can continue training with the loaded model!
+    >>> model = FastText.load(fname)
+
+Once loaded, such models behave identically to those created from scratch.
+For example, you can continue training the loaded model:
+
+    >>> new_sentences = [
+    ...     ['sweet', 'child', 'of', 'mine'],
+    ...     ['rocket', 'queen'],
+    ...     ['you', 'could', 'be', 'mine'],
+    ...     ['november', 'rain'],
+    ... ]
+    >>> 'rocket' in model.wv
+    False
+    >>> model.train(new_sentences, total_examples=len(sentences), epochs=model.epochs)
```

Comment:
Why `total_example=len(sentences)`? Even if correct, it looks strange… my first question as a user.
Let's provide the answer here in the docs.

Reply:
It's correct. I agree that is confusing. The docstring for the [train function](https://radimrehurek.com/gensim/models/fasttext.html#gensim.models.fasttext.FastText.train) attempts to clarify the situation.

Personally, I think if neither total_examples and total_words are specified, we should try to determine sensible defaults by looking at e.g. len(sentences). WDYT @menshikh-iv ?

Reply:
Are you sure? I read the docs and still don't get why it's not `len(new_sentences)`.

Please include some top-level intuition here. A short sentence on why this parameter is mandatory, and what should be its value, because it looks really strange and superfluous.

Reply:
For `train()` to manage `alpha` correctly, and show meaningful progress estimates, it needs a good estimate of the size of the supplied-corpus - even when the corpus (as an iterable) may not self-report its length. In the typical case where the same corpus was just surveyed for its vocabulary, this value should be handy. (In the Word2Vec case, the count from the vocab-scan is cached inside the model for later consultation – unsure if the FT paths do this.) In the case where other/new data is being supplied, the caller should supply the right counts for the current data.

Reply:
If it's able to self-report its length, in count of texts, then yes, that would work as `total_examples`. But because streaming a corpus from an iterable is the motivating case for this interface, relying on that inside the method seems inappropriate to me. Leaving it the responsibility of the caller seems fine to me – and if they're lucky enough to have a corpus that reports its own length, they can supply it easily.

Reply:
Also, while we're talking about simplifying the API, what do you think about removing the sentences and corpus_file parameters from the constructor? Currently, we have an inconsistency: in the constructor, we just pass sentences/corpus_file **without** total_examples and total_words parameters.  In the train function, we include those additional parameters.

Instead of passing sentences in the constructor, the user can pass them in separately via the train function.

Pros:

- Simpler constructor.  There will be less parameters.
- Easier to understand and consistent API.

Cons:

- Getting a trained model now takes two steps instead of one (instantiate, train). Not sure how much of a con this is given that the train function is more powerful than the constructor anyway - it has parameters that the constructor doesn't.

@menshikh-iv @piskvorky @gojomo What do you think?

Reply:
>  But because streaming a corpus from an iterable is the motivating case for this interface, relying on that inside the method seems inappropriate to me.

@gojomo Why do you think it is inappropriate? We could do something like this:

```python
if total_examples or total_words:
    pass  # nothing to do here
elif sentences and hasattr(sentences, '__len__'):
    total_examples = len(sentences)
elif data_corpus and hasattr(data_corpus, '__len__'):
    total_examples = len(data_corpus)
else:
    raise ValueError(
        'unable to infer total_examples or total_words from the training source, '
        'please pass one of them explicitly'
    )
```

It looks ugly, but it allows to user to do something like:

````python
model.train(sentences)
````

instead of

```python
model.train(sentences, num_examples=len(sentences))
```

I feel the former is more Pythonic.

Finally, I think having two separate keyword parameters for the input is confusing for the user. In my opinion, it would look a lot simpler if we unified the two parameters, and dealt with untangling them in the implementation.

Reply:
@mpenkov yes, I'd consider that "sensible" defaults. Thanks.

Agreed on unifying parameters. How would that work? Keep both as acceptable input (internally unify to one), no deprecations? Deprecate one? Which one?

Reply:
OK. I think it's worth dealing with API refactoring in a separate PR, for two reasons:

1. Such changes risk introducing regressions, and I'd rather not have them together with the relatively safe documentation changes.
2. We've already spent much effort refactoring FastText, and it may be prudent to wait for things to stabilize (e.g. fix introduced regressions) before proceeding.